opm/grid/cpgrid/CpGridData.hpp

```diff
@@ -257,6 +285,592 @@ class CpGridData
+        // I_FACES
+        for (int j = startIJK[1]; j < endIJK[1]; ++j) {
+            for (int i = startIJK[0]; i < endIJK[0]+1; ++i) {
+                for (int k = startIJK[2]; k < endIJK[2]; ++k) {
+                    patch_faces.push_back((j*(grid_dim[0]+1)*grid_dim[2]) +(i*grid_dim[2]) + k);
+                } // end k-for-loop
+            } // end i-for-loop
+        } // end j-for-loop
+        // J_FACES
+        for (int j = startIJK[1]; j < endIJK[1]+1; ++j) {
+            for (int i = startIJK[0]; i < endIJK[0]; ++i) {
+                for (int k = startIJK[2]; k < endIJK[2]; ++k) {
+                    patch_faces.push_back(((grid_dim[0]+1)*grid_dim[1]*grid_dim[2]) // i_grid_faces
+                                          + (j*grid_dim[0]*grid_dim[2]) + (i*grid_dim[2]) + k);
+                } // end k-for-loop
+            } // end i-for-loop
+        } // end j-for-loop
+        // K_FACES
+        for (int j = startIJK[1]; j < endIJK[1]; ++j) {
+            for (int i = startIJK[0]; i < endIJK[0]; ++i) {
+                for (int k = startIJK[2]; k < endIJK[2]+1; ++k) {
+                    patch_faces.push_back((grid_dim[0]*(grid_dim[1]+1)*grid_dim[2]) //j_grid_faces
+                                          + ((grid_dim[0]+1)*grid_dim[1]*grid_dim[2])          // i_grid_faces
+                                          + (j*grid_dim[0]*(grid_dim[2]+1)) + (i*(grid_dim[2]+1))+ k);
+                } // end k-for-loop
+            } // end i-for-loop
+        } // end j-for-loop
+        /// PATCH CELLS
+        std::vector<int> patch_cells;
+        patch_cells.reserve(patch_dim[0]*patch_dim[1]*patch_dim[2]);
+        for (int k = startIJK[2]; k < endIJK[2]; ++k) {
+            for (int j = startIJK[1]; j < endIJK[1]; ++j) {
+                for (int i = startIJK[0]; i < endIJK[0]; ++i) {
+                    patch_cells.push_back((k*grid_dim[0]*grid_dim[1]) + (j*grid_dim[0]) +i);
+                } // end i-for-loop
+            } // end j-for-loop
+        } // end k-for-loop
+        return {patch_corners, patch_faces, patch_cells};
+    }
+
+    /// @brief
+    ///   Construct a 'fake cell (Geometry<3,3> object)' out of a patch of cells.(Cartesian grid required).
+    ///
+    /// cellifyPatch() builds a Geometry<3,3> object, 'a celliFIED patch', from a connected patch formed
+    /// by the product of consecutive cells in each direction; selecting 8 corners of the patch boundary,
+    /// computing center and volume.
+    ///
+    /// @param [in] startIJK                 Cartesian triplet index where the patch starts.
+    /// @param [in] endIJK                   Cartesian triplet index where the patch ends.
+    /// @param [in] patch_cells              Cell indices from the block-shaped patch.
+    /// @param [in] cellifiedPatch_geometry  Required as an argument when creating a Geomtry<3,3> object.
+    /// @param [in] cellifiedPatch_to_point  To store the 8 corners of the created cellifiedPatch.
```

Comment:
Should probably be `@param [out]`

Reply:
Got it. When we pass a parameter as an argument that gets populated inside the definition of the function, we use @param [out]. If the functions returns sth, then @return.